src/sort-filter-bar/sort-filter-bar.ts

```diff
@@ -341,46 +390,148 @@ export class SortFilterBar
         href="#"
         @click=${(e: Event) => {
           e.preventDefault();
-          if (options?.clickEvent) {
-            options.clickEvent(e);
-          } else {
-            this.alphaSelectorVisible = null;
-            this.dateSortSelectorVisible = false;
-            this.selectedTitleFilter = null;
-            this.selectedCreatorFilter = null;
-            this.setSelectedSort(sortField);
-            this.emitTitleLetterChangedEvent();
-            this.emitCreatorLetterChangedEvent();
-          }
+          options?.onClick?.(e);
         }}
-        class=${isSelected() ? 'selected' : ''}
+        class=${isSelected() ? 'selected' : nothing}
       >
         ${displayName}
       </a>
     `;
   }
 
-  private get mobileSortSelectorTemplate() {
+  /**
+   * Generates a dropdown component containing multiple grouped sort options.
+   *
+   * @param options.displayName The name to use for the dropdown's visible label
+   * @param options.id The id to apply to the dropdown element
+   * @param options.dropdownOptions An array of option objects used to populate the dropdown
+   * @param options.selectedOption The id of the option that should be initially selected
+   * @param options.isSelected A function returning a boolean indicating whether this dropdown
+   *  should use its selected appearance
+   * @param options.onOptionSelected A handler for optionSelected events coming from the dropdown
+   * @param options.onDropdownClick A handler for click events on the dropdown
+   * @param options.onLabelInteraction A handler for click events and Enter/Space keydown events
+   *  on the dropdown's label
+   */
+  private getSortDropdown(options: {
+    displayName: TemplateResult;
+    id?: string;
+    dropdownOptions: optionInterface[];
+    selectedOption?: string;
+    isSelected?: () => boolean;
+    onOptionSelected?: (e: CustomEvent<{ option: optionInterface }>) => void;
+    onDropdownClick?: (e: PointerEvent) => void;
+    onLabelInteraction?: () => void;
+  }): TemplateResult {
     return html`
-      <select
-        id="mobile-sort-selector"
-        @change=${this.mobileSortChanged}
-        class=${this.mobileSelectorVisible ? 'visible' : 'hidden'}
+      <ia-dropdown
+        id=${options.id ?? nothing}
+        class=${options.isSelected?.() ? 'selected' : nothing}
+        displayCaret
+        closeOnSelect
+        includeSelectedOption
+        .openViaButton=${false}
+        .options=${options.dropdownOptions}
+        .selectedOption=${options.selectedOption}
```

Comment:
SUGGESTION: `selectedOption` could be null, right? So provide default with `??`?

Reply:
Thanks, should be fixed now!